agent/artifact_downloader.go

```diff
@@ -9,6 +9,7 @@ import (
 	"runtime"
 	"strings"
 
+	"github.com/aws/aws-sdk-go/service/s3"
```

Comment:
It's a slight shame the the S3-specific internals leak out to this layer.

I guess the alternative would be an s3_downloader context or wrapped client or something… but that would add complexity and it's not even in a separate package, so all good.